README.md

```diff
@@ -145,7 +166,7 @@ code, include a script or makefile that will do the job.
 
 ## A2.5
 
-«replace this with your answer»
+main.cpp
```

Comment:
Thanks for the set of tests. My only problem with them is that they just output a stream of zeros and ones. They rely on a human to both read them and to match them against the correct results. That's what computers are good at: an automated test should test expected results against actual results, and report when they differ.

# -2

My other problem with the code is that you don't actually have an explicit state machine in there, nor do you have the states represented as a table. You have a parser, but it's really just the same manual code that you'd have written before this class. I was looking for something like this:

~~~ cpp
typedef int state;
typedef std::map<std::string, state> transitions;
typedef std::map<state, transitions> state_table;

const int END = 99;

state_table states = {
  { 0, {{ "the", 1 }}},
	{ 1, {{ "lazy", 2}, {"smelly", 2}, {"dog", 3}, {"cat", 3}}},
	{ 2, {{ "lazy", 2}, {"smelly", 2}, {"dog", 3}, {"cat", 3}}},
	{ 3, {{ "ate", 4},  {"ran",    4}}},
	{ 4, {{ "slowly", 5}, { "noisily", 5}, {"EOI", END}}},
	{ 5, {{ "EOI", END }}}
};


int getNextState(int currentState, string nextWord) {
	transitions t = states[currentState];
	if (t.count(nextWord))
		return t[nextWord];
	else
		return -1;
}
~~~

# -6